client/components/marketing-survey/cancel-purchase-form/step-components/test/business-at-step.jsx

```diff
@@ -1,75 +1,84 @@
+		expect( screen.queryByRole( 'group' ) ).toHaveTextContent(
+			'Translated: Have a theme or plugin you need to install to build the site you want? ' +
+				'Now you can! ' +
+				'Learn more about {{pluginLink}}installing plugins{{/pluginLink}} and ' +
+				'{{themeLink}}uploading themes{{/themeLink}} today.'
+		);
+	} );
 
-		test( 'should render translated confirmation content', () => {
-			expect( wrapper.find( 'FormFieldset > p' ).at( 1 ).props().children ).toEqual(
+	test( 'should render translated confirmation content', () => {
+		render( <BusinessATStep recordTracksEvent={ noop } translate={ translate } /> );
+		expect(
+			screen.getByText(
 				'Translated: Are you sure you want to cancel your subscription and lose access to these new features?'
-			);
-		} );
+			)
+		).toBeVisible();
 	} );
+} );
 
-	describe( 'rendered links', () => {
-		const translate = ( content, params ) => {
-			if ( params && params.components ) {
-				return (
-					<div>
-						{ params.components.pluginLink }
-						{ params.components.themeLink }
-					</div>
-				);
-			}
-			return null;
-		};
-		let recordTracksEvent;
-		let wrapper;
+describe( 'rendered links', () => {
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
 
-		beforeEach( () => {
-			recordTracksEvent = jest.fn();
-			wrapper = shallow(
-				<BusinessATStep translate={ translate } recordTracksEvent={ recordTracksEvent } />
+	const translate = ( content, params ) => {
+		if ( params && params.components ) {
+			return (
+				<>
+					{ params.components.pluginLink }
+					{ params.components.themeLink }
+				</>
 			);
-		} );
+		}
+		return null;
+	};
+	const recordTracksEvent = jest.fn();
 
-		test( 'should fire tracks event for plugin support link when clicked', () => {
-			wrapper.find( 'a' ).at( 0 ).simulate( 'click' );
+	test( 'should fire tracks event for plugin support link when clicked', async () => {
+		render( <BusinessATStep translate={ translate } recordTracksEvent={ recordTracksEvent } /> );
+		const link = screen.queryAllByRole( 'link' )[ 0 ];
 
-			expect( recordTracksEvent ).toHaveBeenCalledWith(
-				'calypso_cancellation_business_at_plugin_support_click'
-			);
-		} );
+		link.addEventListener( 'click', ( event ) => event.preventDefault(), false );
+		await userEvent.click( link );
 
-		test( 'should fire tracks event for theme support link when clicked', () => {
-			wrapper.find( 'a' ).at( 1 ).simulate( 'click' );
+		expect( recordTracksEvent ).toHaveBeenCalledWith(
+			'calypso_cancellation_business_at_plugin_support_click'
+		);
+	} );
 
-			expect( recordTracksEvent ).toHaveBeenCalledWith(
-				'calypso_cancellation_business_at_theme_support_click'
-			);
-		} );
+	test( 'should fire tracks event for theme support link when clicked', async () => {
+		render( <BusinessATStep translate={ translate } recordTracksEvent={ recordTracksEvent } /> );
+		const link = screen.queryAllByRole( 'link' )[ 1 ];
```

Comment:
We could as well use `data-testid` in mocks to be even more specific if you don't want to traverse throughout all the links found.

Reply:
My understanding is that using `data-testid` in production code is not recommended, but when it's in mocks I find it just fine. Querying by role is just fine as well, though, so feel free to keep what you have.

Reply:
Ah, okay - that makes sense! 👍